cpp/leaky_core.cc

```diff
@@ -16,6 +16,25 @@
 
 namespace pyos {
 
+static List<int>* AllocSignalQueue() {
+  List<int>* ret = NewList<int>();
+  ret->reserve(kMaxSignalsInFlight);
+  return ret;
+}
+
+static List<int>* signal_queue = nullptr;
+static int last_sig_num = 0;
+static int sigwinch_num = UNTRAPPED_SIGWINCH;
```

Comment:
I _think_ the writes to these should be safe. If we want to be safe we can mark them as explicitly atomic or word-align them so they get treated similarly to pointers